### ISSUE TITLE
Ported to Haxe. neuron has no field neuron in network.js

### ISSUE THREAD

**Issue by profound7** [cazala/synaptic#116]
Hi. I've made a Haxe port of this library. Haxe is something like TypeScript where variables can be typed, except that it can transpile to multiple other languages instead of just javascript.

While porting, I noticed this code in network.js:
```javascript
var neuron = list[i].neuron;
while (neuron.neuron)
    neuron = neuron.neuron
```
https://github.com/cazala/synaptic/blob/master/src/network.js#L134

When I port it to haxe, it should become like so:
```haxe
var neuron:Neuron = list[i].neuron;
while (neuron.neuron != null)
    neuron = neuron.neuron;
```
https://github.com/profound7/moon-ai/blob/master/src/moon/ai/neural/Network.hx#L263

This produces an error in the haxe compiler that says "neuron has no field neuron".

So, I inspected the type of the variable `list`, and looked for all references of it in your source codes. I concluded that list is an array of an object { neuron, layer } where neuron is a Neuron object and layer is a string.

So in Haxe, list is:
```haxe
var list:Array<NeuronLayerInfo>;
...
typedef NeuronLayerInfo =
{
    var neuron:Neuron;
    var layer:String;
}
```

So doing a `neuron.neuron` is accessing the neuron field of the Neuron object, however, Neuron objects don't have a neuron field, so neuron.neuron is always null, and the while loop won't ever run.

To test this, I commented out all the lines with `while (neuron.neuron) neuron = neuron.neuron`, and the program still works normally as expected.

So are those while loops not needed, or are they required and I misunderstood something?

**Comment by cazala** [comment]
Yup, you are right, that's actually old code... those while loops are never run, we should remove them.
